fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -173,21 +173,25 @@ def app_name
     end
 
     def mac?
-      # Some projects have different values... we have to look for all of them
-      return true if build_settings(key: "PLATFORM_NAME") == "macosx"
-      return true if build_settings(key: "PLATFORM_DISPLAY_NAME") == "macOS"
-      return true if build_settings(key: "PLATFORM_DISPLAY_NAME") == "OS X"
-      false
+      supported_platforms.include?(:macOS)
     end
 
     def tvos?
-      return true if build_settings(key: "PLATFORM_NAME").to_s.include? "appletv"
-      return true if build_settings(key: "PLATFORM_DISPLAY_NAME").to_s.include? "tvOS"
-      false
+      supported_platforms.include?(:tvOS)
     end
 
     def ios?
-      !mac? && !tvos?
+      supported_platforms.include?(:iOS)
+    end
+
+    def supported_platforms
+      supported_platforms = build_settings(key: "SUPPORTED_PLATFORMS").split
+      platforms = []
+      platforms << :macOS   if supported_platforms.include?("macosx")
+      platforms << :iOS     if (supported_platforms & ["iphonesimulator", "iphoneos"]).any?
+      platforms << :watchOS if (supported_platforms & ["watchsimulator", "watchos"]).any?
+      platforms << :tvOS    if (supported_platforms & ["appletvsimulator", "appletvos"]).any?
+      return platforms
     end
```

Comment:
Working around Ruby's lack of Sets is always a pain... What do you think about this:

```ruby
supported_platforms = build_settings(key: "SUPPORTED_PLATFORMS").split
supported_platforms.reduce([]) do |acc, val|
  acc << case val
         when "macosx" then :macOS
         when "iphonesimulator", "iphoneos" then :iOS
         when "watchsimulator", "watchos" then :watchOS
         when "appletvsimulator", "appletvos" then :tvOS
         end
end.uniq!
```